packages/framework/src/types/step.types.ts

```diff
@@ -54,10 +50,24 @@ type StepContext = {
 
 type StepOutput<T_Result> = Promise<T_Result & StepContext>;
 
-export type ActionStep<T_Outputs, T_Result> = (
+export type ActionStep<T_Outputs, T_Result> = <
+  /**
+   * The schema for the inputs of the step.
+   */
+  T_InputSchema extends Schema,
+  /**
+   * The inputs for the step.
+   */
+  T_Inputs = FromSchema<T_InputSchema>
+>(
   name: string,
-  resolve: (inputs: any) => Awaitable<T_Outputs>,
-  options?: ActionStepOptions
+  resolve: (inputs: T_Inputs) => Awaitable<T_Outputs>,
+  options?: {
+    skip?: Skip<T_Inputs>;
+    inputSchema?: T_InputSchema;
+    // TODO: Remove the providers from the action step options
+    providers?: Record<string, (payload: unknown) => Awaitable<unknown>>;
+  }
```

Comment:
This was a fix of the ActionStep typings, previously the `inputs` were always shown as `any` and weren't getting input schema type inference.